### PR TITLE
Support setting testRootPath as a last CLI argument instead of named argument

### LINKED ISSUE #180: Support setting testRootPath as a last CLI argument instead of named argument
After #179 . We should check whether provided directory contains toml file and wether it is root config, then we should look for properties file in the same location

### PR DESCRIPTION
**Pull request by Cheshiriks**
What's done:
* Deleted testRootPath
Closes #180

### CODE REVIEW

File: save-cli/src/nativeMain/kotlin/org/cqfn/save/cli/SaveConfigReader.kt [hunk 1]
```diff
@@ -24,10 +24,18 @@ import kotlinx.serialization.serializer
  * @return this config in case we have valid configuration
  */
 fun SaveProperties.validate(): SaveProperties {
-    this.testRootPath ?: logErrorAndExit(ExitCodes.INVALID_CONFIGURATION,
-        "`test-root-path` option is missing or null. " +
+    this.testFiles ?: logErrorAndExit(ExitCodes.INVALID_CONFIGURATION,
+        "`test-files option is missing or null. " +
```
Comment (akuleshov7) [inline save-cli/src/nativeMain/kotlin/org/cqfn/save/cli/SaveConfigReader.kt:28]: What is test-files option? How do you expect user to understand

File: save-cli/src/nativeMain/kotlin/org/cqfn/save/cli/SaveConfigReader.kt [hunk 2]
```diff
@@ -53,11 +61,27 @@ fun SaveProperties.getFields() = this.toString().dropWhile { it != '(' }.drop(1)
...
+        throwErrorAndExitNotFoundDir()
+    }
     val configFromCli = SaveProperties(args)
     tryToUpdateDebugLevel(configFromCli)
     logDebug("Properties after parsed command line args:\n${configFromCli.getFields()}")
     // reading configuration from the properties file
-    val propertiesFile = (configFromCli.testRootPath ?: defaultConfig().testRootPath) + DIRECTORY_SEPARATOR + "save.properties"
+    val testFiles = configFromCli.testFiles
+    try {
+        if (testFiles != null && testFiles.isNotEmpty()) {
```
Comment (akuleshov7) [inline save-cli/src/nativeMain/kotlin/org/cqfn/save/cli/SaveConfigReader.kt:73]: In kotlin there is a null or empty check

File: save-cli/src/nativeMain/kotlin/org/cqfn/save/cli/SaveConfigReader.kt [hunk 3]
```diff
@@ -53,11 +61,27 @@ fun SaveProperties.getFields() = this.toString().dropWhile { it != '(' }.drop(1)
...
+            FileSystem.SYSTEM.metadata(testFiles[0].toPath())
+        }
+    } catch (e: FileNotFoundException) {
+        throwErrorAndExitNotValidDir(testFiles!!.get(0).toPath())
+    }
+    val testRootPath = if (testFiles == null || testFiles.isEmpty() || !FileSystem.SYSTEM.metadata(testFiles!!.get(0).toPath()).isDirectory) {
```
Comment (akuleshov7) [inline save-cli/src/nativeMain/kotlin/org/cqfn/save/cli/SaveConfigReader.kt:79]: Why get(0)

Comment (akuleshov7) [reply save-cli/src/nativeMain/kotlin/org/cqfn/save/cli/SaveConfigReader.kt:79]: i do not like logic on the exceptions. Does okio has any method like `exists` for Path

File: save-cli/src/nativeMain/kotlin/org/cqfn/save/cli/SaveConfigReader.kt [hunk 4]
```diff
@@ -53,11 +61,27 @@ fun SaveProperties.getFields() = this.toString().dropWhile { it != '(' }.drop(1)
...
+        null
+    } else {
+        testFiles[0]
+    }
+    val propertiesFile = testRootPath + DIRECTORY_SEPARATOR + "save.properties"
```
Comment (akuleshov7) [inline save-cli/src/nativeMain/kotlin/org/cqfn/save/cli/SaveConfigReader.kt:84]: Why not to concatenate paths

File: save-cli/src/nativeMain/kotlin/org/cqfn/save/cli/SaveConfigReader.kt [hunk 5]
```diff
@@ -105,3 +129,15 @@ fun readPropertiesFile(propertiesFileName: String?): SaveProperties {
 private fun tryToUpdateDebugLevel(properties: SaveProperties) {
     isDebugEnabled = properties.debug ?: false
 }
+
+private fun throwErrorAndExitNotFoundDir() {
+    logErrorAndExit(ExitCodes.INVALID_CONFIGURATION,
+        "Save expected to get the root directory for test files as the last CLI argument. " +
```
Comment (akuleshov7) [inline save-cli/src/nativeMain/kotlin/org/cqfn/save/cli/SaveConfigReader.kt:135]: ExpectS

**Review by akuleshov7 — CHANGES_REQUESTED** [review]
Remove code smells

File: save-cli/src/nativeMain/kotlin/org/cqfn/save/cli/SaveConfigReader.kt (hunk 3 above)
Comment (Cheshiriks) [reply save-cli/src/nativeMain/kotlin/org/cqfn/save/cli/SaveConfigReader.kt]: Fixed

File: save-cli/src/nativeMain/kotlin/org/cqfn/save/cli/SaveConfigReader.kt (hunk 2 above)
Comment (Cheshiriks) [reply save-cli/src/nativeMain/kotlin/org/cqfn/save/cli/SaveConfigReader.kt]: Fixed

File: save-cli/src/nativeMain/kotlin/org/cqfn/save/cli/SaveConfigReader.kt [hunk 6]
```diff
@@ -46,18 +55,42 @@ fun SaveProperties.validate(): SaveProperties {
 fun SaveProperties.getFields() = this.toString().dropWhile { it != '(' }.drop(1)
     .dropLast(1)
 
+/**
+ * @return true if path is exists
+ */
+public fun Path.exists() = try {
```
Comment (akuleshov7) [inline save-cli/src/nativeMain/kotlin/org/cqfn/save/cli/SaveConfigReader.kt:61]: move to FileUtils

**Review by akuleshov7 — APPROVED** [review]

File: save-cli/src/nativeMain/kotlin/org/cqfn/save/cli/SaveConfigReader.kt (hunk 6 above)
Comment (Cheshiriks) [reply save-cli/src/nativeMain/kotlin/org/cqfn/save/cli/SaveConfigReader.kt:61]: Fixed

**Pull request merged**